cpp/src/arrow/compute/kernels/vector_selection.cc

```diff
@@ -530,6 +683,54 @@ Status PrimitiveTake(KernelContext* ctx, const ExecSpan& batch, ExecResult* out)
   return Status::OK();
 }
 
+Status ChunkedPrimitiveTake(KernelContext* ctx, const ExecBatch& batch, Datum* out) {
+  const ChunkedArray& values = *batch[0].chunked_array();
+  const ChunkedArray& indices = *batch[1].chunked_array();
+
+  if (TakeState::Get(ctx).boundscheck) {
+    RETURN_NOT_OK(CheckIndexBounds(indices, values.length()));
+  }
+
+  // TODO: Is there any reason to chunk the output for primitive arrays?
+  // We probably want to keep within 32-bit sizes for interoperability with other
+  // implementations.
```

Comment:
I'd welcome some design input on this. How should the output chunking be structured?

Reply:
My jerk reaction is to prefer the existing chunk layouts, just in case the chunking was specially chosen for hardware or data placement reasons.

Reply:
So, use the chunking of the take indices? I suppose that makes sense.

Reply:
My intuition would be not to chunk at all. This is what we usually do for vector kernels when there is no natural mapping from input chunking to output chunking.

Reply:
I think the original motivation here, though, was for things that won't fit in a chunk (not necessarily due to number of rows, due to things like string data)

Reply:
Possibly, though there's a more general performance concern (you don't want to concatenate the chunks of a large chunked array just to take 10 elements out of it).

Reply:
(also this comment is about primitive arrays)

Reply:
Ah, I missed that for primitive arrays.

Reply:
This discussion seems like it might be relevant actually: https://issues.apache.org/jira/browse/ARROW-2532